gc/src/lib.rs

```diff
@@ -30,17 +32,13 @@ pub use gc::force_collect;
 ///
 /// See the [module level documentation](./) for more details.
 pub struct Gc<T: Trace + ?Sized + 'static> {
-    // XXX We can probably take advantage of alignment to store this
-    root: Cell<bool>,
-    _ptr: NonZero<*mut GcBox<T>>,
+    _ptr: UnsafeCell<NonZero<*mut GcBox<T>>>,
```

Comment:
Are we sure that it's safe to use both NonZero and fiddle with the low bit at once? Looks safe, but it might become unsafe after optimizations.